core/app/[locale]/(default)/product/[slug]/_components/search-params-router-refresh.tsx

```diff
@@ -0,0 +1,28 @@
+'use client';
+
+import { useSearchParams } from 'next/navigation';
+import { SearchParams } from 'nuqs';
+import { useEffect } from 'react';
+
+import { useRouter } from '~/i18n/routing';
+
+// Not-so-great workaround for https://github.com/vercel/next.js/issues/59407
```

Comment:
Honestly, maybe we should disable PPR since it flashes when there's a change. Maybe @bookernath and @migueloller can provide an opinion.